### PR TITLE
Refactor access modals P/E/T [SCI-7619]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-7619](https://scinote.atlassian.net/browse/SCI-7619)

### What was done
Refactor access modals

[SCI-7619]: https://scinote.atlassian.net/browse/SCI-7619?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/controllers/access_permissions/experiments_controller.rb [hunk 1]
```diff
@@ -20,14 +20,33 @@ def edit
     end
 
     def update
-      @experiment_member = ExperimentMember.new(current_user, @experiment, @project)
+      user_id = permitted_update_params[:user_id]
+      @user_assignment = UserAssignment.find_by(
+        assignable: @experiment,
+        user_id: user_id
+      )
```
Comment (artoscinote) [inline app/controllers/access_permissions/experiments_controller.rb:27]: I'm pretty sure we need to also include team_id here.

File: app/controllers/access_permissions/projects_controller.rb [hunk 2]
```diff
@@ -108,12 +123,42 @@ def set_project
...
+
     def check_manage_permissions
       render_403 unless can_manage_project_users?(@project)
     end
 
     def check_read_permissions
       render_403 unless can_read_project?(@project)
     end
+
+    def available_users
```
Comment (artoscinote) [inline app/controllers/access_permissions/projects_controller.rb:143]: could possibly be moved to the assignable concern, as `assignable_users`

File: app/controllers/access_permissions/experiments_controller.rb [hunk 3]
```diff
@@ -20,14 +20,33 @@ def edit
...
 
       if permitted_update_params[:user_role_id] == 'reset'
-        @experiment_member.reset(permitted_update_params)
+        @user_assignment.update(
```
Comment (artoscinote) [inline app/controllers/access_permissions/experiments_controller.rb:30]: `.update!`

**Review by artoscinote — CHANGES_REQUESTED** [review]

File: app/controllers/access_permissions/projects_controller.rb (hunk 2 above)
Comment (aignatov-bio) [reply app/controllers/access_permissions/projects_controller.rb:143]: This logic is dedicated to projects

**Review by artoscinote — APPROVED** [review]

**Pull request merged**